action.yml

```
@@ -0,0 +1,140 @@
+name: Gauge release
+description: Gauge if should do a patch release
+
+inputs:
+  latest_local_sha:
+    description: The latest local sha
+    required: true
+    type: string
+
+outputs:
+  do_release:
+    description: Whether to do a release
+    value: ${{ steps.gauge-release.outputs.do_release }}
+  next_tag:
+    description: The patch tag to release
+    value: ${{ steps.gauge-release.outputs.next_tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Gauge release
+      id: gauge-release
+      shell: bash
+      env:
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        GITHUB_REF_NAME: ${{ github.ref_name }}
+        GITHUB_SHA: ${{ github.sha }}
+        LATEST_LOCAL_SHA: ${{ inputs.latest_local_sha }}
+      run: |
+        DO_RELEASE=1
+
+        # Double check that LATEST_LOCAL_SHA matches GITHUB_SHA
+        if [[ $DO_RELEASE == "1" ]]; then
+          if [[ $LATEST_LOCAL_SHA != $GITHUB_SHA ]]; then
+            echo "LATEST_LOCAL_SHA $LATEST_LOCAL_SHA does not match GITHUB_SHA $GITHUB_SHA"
+            DO_RELEASE=0
+          fi
+        fi
```

```suggestion
        if [[ $LATEST_LOCAL_SHA != $GITHUB_SHA ]]; then
          echo "LATEST_LOCAL_SHA $LATEST_LOCAL_SHA does not match GITHUB_SHA $GITHUB_SHA"
          DO_RELEASE=0
        fi
```
`$DO_RELEASE` can't be anything other than "1" at this point.

---

Updated